sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -351,6 +352,12 @@ public <T> T findById(Object id, Class<T> domainType, PartitionKey partitionKey)
     public <T> T findById(String containerName, Object id, Class<T> domainType) {
         Assert.hasText(containerName, "containerName should not be null, empty or only whitespaces");
         Assert.notNull(domainType, "domainType should not be null");
+        String containerPartitionKey = getContainerProperties(containerName).getPartitionKeyDefinition()
+            .getPaths().iterator().next().replaceAll("^/|/$", "");
+        if (containerPartitionKey.equals("id")) {
```

Comment:
I believe this will have some issues. You can make any field the Id by adding @Id above it so there is no guarantee that the name of the id field will be "id". You will want to create a CosmosEntityInformation objet and use the getIdFieldName() function instead of just putting in "id". Message me if you need me to show you how to do this :)

Reply:
Here we’re looking for the field that has been defined as the partition key for the container. It’s true you can define any field as “Id” but that doesn’t dictate what has been defined as the partitionKey from my understanding.

Reply:
So if I define “lastName” as id, as far as I understand the value of “lastName” will be used to populate “id”. But if I’ve said “id” is the @partitionKey, then the result of this assignment will still be “id”, and that’s what’s being checked here. What’s crucial is whether Id has been defined as partitionKey. One thing I don’t know is whether id is defined as partitionKey by default if nothing is defined for partition key. Will have to test.

Reply:
I think it is worth testing, if you have:

@Id
@PartitionKey
String lastName;

It may just be putting lastName in a field called "id" but I don't think that is the case, I think "lastName" is the only field created but even if it is put in field "id" I still think when you get the name of partition key it will be "lastName".  A lot of room for things to regress regardless with so many different pieces so worth adding the test where the field is not named "id" if you ask me.

Reply:
Discussed offline with Theo, no action needed.